test/compiler/axis.spec.js

```diff
@@ -36,4 +42,134 @@ describe('Axis', function() {
+    // FIXME(kanitw): Jul 19, 2015 - write test
+  });
+
+  describe('orient()', function () {
+    it('should return specified orient', function () {
+      var orient = axis.orient('x', Encoding.fromSpec({
+          encoding: {
+            x: {name: 'a', axis:{orient: 'bottom'}}
+          }
+        }), stats);
+      expect(orient).to.eql('bottom');
+    });
+
+    it('should return undefined by default', function () {
+      var orient = axis.orient('x', Encoding.fromSpec({
+          encoding: {
+            x: {name: 'a'}
+          }
+        }), stats);
+      expect(orient).to.eql(undefined);
+    });
+
+    it('should return top for COL', function () {
+      var orient = axis.orient('col', Encoding.fromSpec({
+          encoding: {
+            x: {name: 'a'},
+            col: {name: 'a'}
+          }
+        }), stats);
+      expect(orient).to.eql('top');
+    });
+
+    it('should return top for X with high cardinality, ordinal Y', function () {
+      var orient = axis.orient('x', Encoding.fromSpec({
+          encoding: {
+            x: {name: 'a'},
+            y: {name: 'b', type: 'O'}
+          }
+        }), stats);
+      expect(orient).to.eql('top');
+    });
+  });
+
+  describe('title()', function () {
+    it('should add explicitly specified title', function () {
+      var def = axis.title({}, 'x', Encoding.fromSpec({
+          encoding: {
+            x: {name: 'a', axis: {title: 'Custom'}}
+          }
+        }), stats, layout);
+      expect(def.title).to.eql('Custom');
+    });
+
+    it('should add return fieldTitle by default', function () {
+      var encoding = Encoding.fromSpec({
+          encoding: {
+            x: {name: 'a', axis: {titleMaxLength: '3'}}
+          }
+        });
+
+      var def = axis.title({}, 'x', encoding, layout);
+      expect(def.title).to.eql('a');
+    });
+
+    it('should add return fieldTitle by default', function () {
+      var encoding = Encoding.fromSpec({
+          encoding: {
+            x: {name: 'a', aggregate: 'sum', axis: {titleMaxLength: '10'}}
+          }
+        });
+
+      var def = axis.title({}, 'x', encoding, layout);
+      expect(def.title).to.eql('SUM(a)');
+    });
+
+    it('should add return fieldTitle by default and truncate', function () {
+      var encoding = Encoding.fromSpec({
+          encoding: {
+            x: {name: 'a', aggregate: 'sum', axis: {titleMaxLength: '3'}}
+          }
+        });
+
+      var def = axis.title({}, 'x', encoding, layout);
+      expect(def.title).to.eql('SU…');
+    });
+
+
+    it('should add return fieldTitle by default and truncate', function () {
+      var encoding = Encoding.fromSpec({
+          encoding: {
+            x: {name: 'abcdefghijkl'}
+          }
+        });
+
+      var def = axis.title({}, 'x', encoding, layout);
+      expect(def.title).to.eql('abcdefghi…');
+    });
+  });
+
+  describe('titleOffset()', function () {
+    // FIXME(kanitw): Jul 19, 2015 - write test
```

Comment:
Krist suggest me that writing data in TODOs/FIXMEs is a good forcing function to make you do it sooner lol.